common/hyperscaler/rules/model/config_test.go

```diff
@@ -0,0 +1,43 @@
+package model
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoad(t *testing.T) {
+	t.Run("should load rules from a file", func(t *testing.T) {
+
+		// given
+		expectedRules := []string{"rule1", "rule2"}
+		content := "rule:\n"
+
+		for _, rule := range expectedRules {
+			content += "- " + rule + "\n"
+		}
+		tmpfile, err := os.CreateTemp("", "test*.yaml")
+		require.NoError(t, err)
+		defer os.Remove(tmpfile.Name())
+
+		if _, err := tmpfile.Write([]byte(content)); err != nil {
+			t.Fatalf("Failed to write to temp file: %v", err)
+		}
+		if err := tmpfile.Close(); err != nil {
+			t.Fatalf("Failed to close temp file: %v", err)
+		}
+
+		// when
+		var config RulesConfig
+		_, err = config.Load(tmpfile.Name())
```

Comment:
And here Load() signature make it harder to get. You skip the return value so at first glimpse we ignore the result of computations apart from error. But we change the state of the config variable.